File: zenwave-jpa-example/src/main/java/io/zenwave360/example/core/implementation/mappers/EventsMapper.java
package io.zenwave360.example.core.implementation.mappers;

import io.zenwave360.example.core.domain.Address;
import io.zenwave360.example.core.domain.Customer;
import io.zenwave360.example.core.domain.PaymentMethod;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(uses = { BaseMapper.class })
public interface EventsMapper {

    EventsMapper INSTANCE = Mappers.getMapper(EventsMapper.class);

    io.zenwave360.example.core.outbound.events.dtos.Address asAddress(Address address);

    io.zenwave360.example.core.outbound.events.dtos.PaymentMethod asPaymentMethod(
            PaymentMethod paymentMethod);

    io.zenwave360.example.core.outbound.events.dtos.CustomerEvent asCustomerEvent(
            Customer customer);

    io.zenwave360.example.core.outbound.events.dtos.CustomerEvent asCustomerEvent(Long id);

}

File: examples/kustomer-address-jpa/src/main/kotlin/io/zenwave360/examples/kotlin/service/impl/mappers/EventsMapper.kt
package io.zenwave360.examples.kotlin.service.impl.mappers

import io.zenwave360.examples.kotlin.service.impl.mappers.BaseMapper
import io.zenwave360.examples.kotlin.domain.*
import io.zenwave360.examples.kotlin.service.dtos.*

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers


@Mapper(uses = [BaseMapper::class])
interface EventsMapper {

    companion object {
        val INSTANCE: EventsMapper = Mappers.getMapper(EventsMapper::class.java)
    }
    fun asAddress(address: Address): io.zenwave360.examples.kotlin.events.dtos.Address
    fun asPaymentMethod(paymentMethod: PaymentMethod): io.zenwave360.examples.kotlin.events.dtos.PaymentMethod
    fun asCustomerEvent(customer: Customer): io.zenwave360.examples.kotlin.events.dtos.CustomerEvent
    fun asCustomerEvent(id: Long): io.zenwave360.examples.kotlin.events.dtos.CustomerEvent {
        return io.zenwave360.examples.kotlin.events.dtos.CustomerEvent().withId(id)
    }
}
